lib/src/package_analysis/lower_bound_constraint_analysis.dart

```diff
@@ -0,0 +1,630 @@
+import 'dart:io';
+
+import 'package:analyzer/dart/analysis/analysis_context_collection.dart';
+import 'package:analyzer/dart/analysis/results.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:analyzer/exception/exception.dart';
+import 'package:collection/collection.dart';
+import 'package:pana/src/package_analysis/package_analysis.dart';
+import 'package:pana/src/package_analysis/shapes.dart';
+import 'package:pana/src/package_analysis/shapes_ext.dart';
+import 'package:pana/src/package_analysis/summary.dart';
+import 'package:path/path.dart' as path;
+import 'package:pub_semver/pub_semver.dart';
+import 'package:source_span/source_span.dart';
+
+import 'common.dart';
+
+/// Analyze the target package [targetName], using the temporary directory
+/// [tempPath] to perform analysis, and return a [List] of any found issues -
+/// where a symbol usage cannot be found in the relevant dependency's
+/// [PackageShape].
+///
+/// If [cachePath] is provided, this folder is used to store cached package
+/// metadata. If [pubHostedUrl] is provided, it is used instead of `'pub.dev'`
+/// to retrieve package version information.
+Future<List<LowerBoundConstraintIssue>> lowerBoundConstraintAnalysis({
+  required String targetName,
+  required String tempPath,
+  String? cachePath,
+  String? pubHostedUrl,
+}) async {
+  final dummyPath = path.join(tempPath, 'target');
+  final dependencyFolder = path.join(tempPath, 'dependencies');
+
+  try {
+    await fetchUsingDummyPackage(
+      name: targetName,
+      version: (await fetchSortedPackageVersionList(
+        packageName: targetName,
+        cachePath: cachePath,
+        pubHostedUrl: pubHostedUrl,
+      ))
+          .last,
+      destination: dummyPath,
+      wipeTarget: true,
+      pubHostedUrl: pubHostedUrl,
+    );
+  } on ProcessException catch (exception) {
+    throw AnalysisException(
+        'Failed to download target package $targetName with error code ${exception.errorCode}: ${exception.message}');
```

Comment:
Because we are downloading packages using a dummy, we have no package metadata other than the name of the package when executing `lbcanalysis` - when the SDK constraints are too low, `pub get` fails here.

Maybe we can query an endpoint like https://pub.dev/api/packages/retry BEFORE spawning the `lbcanalysis` process, and only spawn this process if two conditions are met:
1) the lower bound sdk constraint is >= 2.12.0
2) the current sdk version satisfies the sdk constraint

otherwise, do not launch `lbcanalysis` and skip onto the next package

although I'm not sure whether making ~30,000 http requests over the course of ~8 hours (between 1 and 2 requests per second) is OK, with no cache in place. though making one GET request is certainly faster than: launching analysis, writing the pubspec to disk and seeing whether `pub get` succeeds/fails